signer/signer.go

```diff
@@ -146,3 +146,26 @@ func (s *genericSigner) Sign(ctx context.Context, desc ocispec.Descriptor, opts
 	// TODO: re-enable timestamping https://github.com/notaryproject/notation-go/issues/78
 	return sig, &envContent.SignerInfo, nil
 }
+
+// SignBlob signs the descriptor returned by blobGen and returns the marshalled envelope
+func (s *GenericSigner) SignBlob(ctx context.Context, blobGen notation.BlobDescriptorGenerator, opts notation.SignerSignOptions) ([]byte, *signature.SignerInfo, error) {
+	logger := log.GetLogger(ctx)
+	logger.Debugf("Generic blob signing for signature media type %v", opts.SignatureMediaType)
+
+	ks, err := s.KeySpec()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	digestAlg, ok := algorithms[ks.SignatureAlgorithm().Hash()]
+	if !ok {
+		return nil, nil, fmt.Errorf("unknown hashing algo %v", ks.SignatureAlgorithm().Hash())
+	}
```

Comment:
It seems we can refactor this part as it has been used by both `plugin.go` and `signer.go`.

Reply:
But the obtained `KeySpec` instances are all of `signature.KeySpec` type. We can refactor from this point.